servlet/src/main/scala/org/http4s/servlet/Http4sServlet.scala

```diff
@@ -83,15 +83,21 @@ abstract class Http4sServlet[F[_]](
   ): F[Unit] =
     F.delay {
       servletResponse.setStatus(response.status.code)
-      for (header <- response.headers.headers if header.name != ci"Transfer-Encoding")
+      for (header <- response.headers.headers if header.name != `Transfer-Encoding`.name)
         servletResponse.addHeader(header.name.toString, header.value)
     }.attempt
       .flatMap {
         case Right(()) => bodyWriter(response)
         case Left(t) =>
-          response.body.drain.compile.drain.handleError { t2 =>
-            logger.error(t2)("Error draining body")
-          } *> F.raiseError(t)
+          response.entity match {
+            case Entity.Default(body, _) =>
+              body.drain.compile.drain.handleError { t2 =>
```

Comment:
By the way, we can remove the first `drain`, since it has the same meaning (discarding outputs) as the first one.